modules/frontend/frontend.go

```diff
@@ -73,7 +73,7 @@ func NewTripperware(cfg Config, logger log.Logger, registerer prometheus.Registe
 
 			resp, err := rt.RoundTrip(r)
 
-			if resp.StatusCode == http.StatusOK && marshallingFormat == util.JSONTypeHeaderValue {
+			if resp != nil && resp.StatusCode == http.StatusOK && marshallingFormat == util.JSONTypeHeaderValue {
```

Comment:
in what situations is resp nil? have we ever seen this panic?

Reply:
Yep, just panic'ed today. `resp` is nil for a 5xx error.